lib/package_unpack.c

```diff
@@ -387,8 +390,9 @@ unpack_archive(struct xbps_handle *xhp,
 					}
 					rv = 0;
 				} else {
-					rv = xbps_file_hash_check_dictionary(
-					    xhp, binpkg_filesd, "files", buf);
+					xbps_dbg_printf(xhp, "%s: %s\n", entry_pname, file->sha256);
+					assert(file->sha256);
```

Comment:
Might be nice to make it clear that a `(null)` there is an error? Not sure.

`file->sha256 ? file->sha256 : "error..."`